support/util/performanceprofilestatusconfigmap/performanceprofilestatusconfigmap.go

```diff
@@ -0,0 +1,74 @@
+package performanceprofilestatusconfigmap
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	performanceprofilev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	"github.com/openshift/hypershift/support/util/performanceprofilestatus"
+)
+
+func New(controlPlaneNamespace, userClustersNamespace, nodePoolName string, opts ...func(cm *corev1.ConfigMap) error) (*corev1.ConfigMap, error) {
+	cm := getTestPerformanceProfileStatusConfigMap(controlPlaneNamespace, nodePoolName)
+	for _, opt := range opts {
+		if err := opt(cm); err != nil {
+			return nil, err
+		}
+	}
+	return cm, nil
+}
+
+func getTestPerformanceProfileStatusConfigMap(controlPlaneNamespace, nodePoolName string) *corev1.ConfigMap {
```

Comment:
If this is part of a test component, I recommend to separate the `getTestPerformanceProfileStatusConfigMap` from `getPerformanceProfileStatusConfigMap` keeping the last one here and putting the tests things in the (as @enxebre mentioned before too (sorry to repeat myself again :/) to the appropriate `_test.go` file.

Reply:
Do you mean to seperate `getTestPerformanceProfileStatusConfigMap` from the function `New`?

Reply:
most of the `performanceprofilestatusconfigmap` and `performanceprofilestatus` packages are used to create mocked configmaps (in a real cluster, generated by NTO) that will hold performance profile status.

Reply:
No, I mean, things like:

```
lastHeartbeatTime := "2024-04-18T06:55:45Z"
```

Are meant to be in a `xxx_test.go` file because they are not relevant for the logic of the code, but on testing circumstances.

Reply:
@jparrill anyways I have changed the code please look